### ISSUE TITLE
Handle protocols that start as plain text and switch to TLS

### ISSUE THREAD

**Issue by jtpereyda** [mitmproxy/mitmproxy#4215]
#### Is your feature request related to a problem? Please describe.
I am trying to MITM a MySQL protocol connection. This protocol can start as plain text, and then switch to TLS (as signified by a Client Hello).

#### Describe the solution you'd like
A mode that repeatedly checks messages for a TLS initialization. This does not apply to HTTP as far as I know, and would only be for other TCP protocols.

#### Describe alternatives you've considered
I'm not aware of any alternatives possible with mitmproxy. The next choice I know of is a custom tool.

#### Additional context
I'd also be happy to apply a manual patch for my current purposes if that's feasible.

**Comment by mhils** [comment]
Yeah, supporting STARTTLS etc would be nice. That's probably more important once we support server greetings, and should be trivial to implement for sans-io. :)

refs #1775

**Comment by K-techie** [comment]
Any update on its implementation yet ?

**Comment by glennzw** [comment]
This would be awesome.

**Comment by glennzw** [comment]
Every few months I bump into debugging a `starttls` session and eventually Google my way back here. I'd love this feature to be added (maybe a sponsored feature?).

**Comment by mhils** [comment]
If someone wants to sponsor this I'm happy to talk, otherwise I don't have a use case myself for this at the moment. :)

**Comment by glennzw** [comment]
If anyone wants to chip in gathering some funds to sponsor this feature drop me an email

@jtpereyda @karano-techie @guyskk

**Comment by mhils** [comment]
I appreciate you taking the initiative here @glennzw (and thanks for the nice meeting earlier), but if someone wants to sponsor work on mitmproxy please reach out to me or someone else from @mitmproxy/devs directly. :)

(I absolutely understand there's no malice intended from your side here)

**Comment by sudowoodo200** [comment]
Is there any more interest for this? Would love to take a stab at it.

**Comment by ThaDaVos** [comment]
Any update on this? Currently in the need of a tool which can let me look into the communication between my application and Microsoft SMTP server `smtp.office365.com` when connected through STARTTLS
